### ISSUE TITLE
Beta 3.0.0.5/v2.20.0 - Install Error, RPI Zero 2W. Node 16.14.2 Stopping RA Installing

### ISSUE THREAD

**Issue by townsmcp** [mKeRix/room-assistant#1113]
**Describe the bug**
Cant install RA beta 3.0.0.5 on RPI Zero 2W

**To reproduce**

1. Wipe SD card
2. Install Raspberry PI OS Lite (64bit)
3. Perform full update:
    ```
    sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get dist-upgrade -y && sudo reboot
    ```
5. Install NodeJS 16:
    ```
   curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -
   sudo apt-get install -y nodejs
    ```
6. Install dependencies:
    ```
    sudo apt-get update && sudo apt-get install build-essential libavahi-compat-libdnssd-dev libsystemd-dev bluetooth libbluetooth-dev libudev-dev libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev librsvg2-dev
    ```
7. Try to install RA:
    ```
    sudo npm i --global --unsafe-perm --verbose --foreground-scripts room-assistant@3.0.0-beta.5
    ```
8. Observe failure to install. See attached log file:
[RPI.log](https://github.com/mKeRix/room-assistant/files/8312325/RPI.log)

**Expected behavior**
RA installs

**Environment**
 - room-assistant version: 3.0.0.beta-5
 - installation type: NodeJS
 - hardware: Raspberry Pi Zero 2W
 - OS: Raspberry Pi OS Lite 64

**Comment by townsmcp** [comment]
3 options to try:

1. Follow the RPI zero W install guide rather than RPI 3/4 but force 3.0.0.beta-5
2. Install beta 5 without performing `sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get dist-upgrade -y ` to rule out RPI updates causing issue since installing beta 4 a week ago worked fine.
3. Install beta 4 with full `sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get dist-upgrade -y `

**Comment by townsmcp** [comment]
Option 1 fail. When installing RA, getting an error message:
```
pi@diningroomRA:~ $ sudo npm i --global --unsafe-perm room-assistant@3.0.0-beta.5
sudo: npm: command not found
pi@diningroomRA:~ $ 
```
Option 3 failed. This appears to be an issue due to base image update in RPI.
Option 2 (following RPI 3/4 install) with beta 5 but no upgrade/dist-upgrade performed. Install failed

Final option to try - wipe SD, follow RPI 3/4 install, no upgrade/dist-upgrade, install beta 4

**Comment by townsmcp** [comment]
Final update - all options have failed. Same error as OP. Im unsure why, even trying to install beta 4 again, there is a failure. The final option was exactly how I built the RPI zero 2w a week ago. It appears the issue is with RA install?
```
npm verb unfinished npm timer reify 1647821574559
npm verb unfinished npm timer reify:unpack 1647821653140
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/bintrees 1647821653170
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/swagger-ui-dist 1647821653176
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/moment-timezone 1647821653295
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/systeminformation 1647821653349
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/mathjs 1647821653356
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/lodash 1647821653357
npm verb unfinished npm timer reifyNode:node_modules/room-assistant/node_modules/js-yaml 1647821653358
npm verb code 1
```

I have even tried the new release 2.20.0 and getting the same issue.
Open to suggestions and help

**Comment by townsmcp** [comment]
Issue resolved. The problem is due to the instructions for installing latest version of NodeJS (currently this is 16.14.2). Last week, the installed version of NodeJS was 16.14.0 which worked with RA install. 
Rather than installing Node 16, I installed the latest Node 17:
```
curl -sL https://deb.nodesource.com/setup_17.x | sudo -E bash -
```
rather than:
```
curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -
```

To be clear, the installed version of 17.7.2 is working with RA. @mKeRix not sure if you would want to update the instructions at https://www.room-assistant.io/guide/quickstart-pi.html#installing-raspbian ?

**Comment by idantene** [comment]
> curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -

Have you tried with other node versions?
When trying to install RA with npm, I get the warning:
```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'room-assistant@2.20.0',
npm WARN EBADENGINE   required: { node: '>= 12 < 17' },
npm WARN EBADENGINE   current: { node: 'v20.10.0', npm: '10.2.3' }
npm WARN EBADENGINE }
```

So I'm inclined to install node < 17, but according to https://endoflife.date/nodejs support for node 16 has ended, node 17 isn't even listed, etc.
I wonder if @mKeRix has time to update the npm code? The repo has been awfully quiet...


Edit FWIW (future users): I ended up going with node 16 and friends, even though it's no longer supported.
Additionally, the install hangs on something like `reify:mathjs:... bluetooth-hci-socket` or `reify:maths:... canvas`. I believe it's to do with the hardware.
Somehow I managed to install it on one of the rpiz, so I literally ended up zipping the contents of `/usr/lib/node_modules/room-assistant` and copying it to my other pi zero 2s.
A short link was needed to make sure the CLI `room-assistant` works as expected, but it seems to work from there.

Maybe a similar zip/installation process can be provided here for the different pis, since compilation takes so long and it can easily time out or throttle...
